src/lib/type_cast.hpp

```diff
@@ -40,14 +40,14 @@ const T& get(const AllTypeVariant& value) {
 template <typename T, typename U,
           typename = std::enable_if_t<std::is_constructible_v<std::decay_t<T>, std::decay_t<U>>>>
 inline __attribute__((always_inline)) T type_cast(U&& value) {
-  return T{std::forward<T>(value)};
+  return static_cast<T>(std::forward<U>(value));
```

Comment:
Just to be sure that's not a typo: the change from `std::forward<T>` to `std::forward<U>` was done on purpose?

Reply:
Correct. We need to forward the original type into the constructor.